src/main/scala/com/databricks/spark/sql/perf/Benchmark.scala

```diff
@@ -122,238 +103,14 @@ abstract class Benchmark(
-      }
-
-      val timestamp = System.currentTimeMillis()
-      val resultPath = s"$resultsLocation/timestamp=$timestamp"
-      val combinations = cartesianProduct(variations.map(l => (0 until l.options.size).toList).toList)
-      val resultsFuture = Future {
-
-        // If we're running queries, create tables for them
-        executionsToRun
-          .collect { case query: Query => query }
-          .flatMap { query =>
-            try {
-              query.newDataFrame().queryExecution.logical.collect {
-                case UnresolvedRelation(t, _) => t.table
-              }
-            } catch {
-              // ignore the queries that can't be parsed
-              case e: Exception => Seq()
-            }
-          }
-          .distinct
-          .foreach { name =>
-            try {
-              sqlContext.table(name)
-              logMessage(s"Table $name exists.")
-            } catch {
-              case ae: Exception =>
-                val table = allTables
-                  .find(_.name == name)
-                if (table.isDefined) {
-                  logMessage(s"Creating table: $name")
-                  table.get.data
-                    .write
-                    .mode("overwrite")
-                    .saveAsTable(name)
-                } else {
-                  // the table could be subquery
-                  logMessage(s"Couldn't read table $name and its not defined as a Benchmark.Table.")
-                }
-            }
-          }
-
-        // Run the benchmarks!
-        val results: Seq[ExperimentRun] = (1 to iterations).flatMap { i =>
-          combinations.map { setup =>
-            val currentOptions = variations.asInstanceOf[Seq[Variation[Any]]].zip(setup).map {
-              case (v, idx) =>
-                v.setup(v.options(idx))
-                v.name -> v.options(idx).toString
-            }
-            currentConfig = currentOptions.map { case (k,v) => s"$k: $v" }.mkString(", ")
-
-            val result = ExperimentRun(
-              timestamp = timestamp,
-              iteration = i,
-              tags = currentOptions.toMap ++ tags,
-              configuration = currentConfiguration,
-
-              executionsToRun.flatMap { q =>
-                val setup = s"iteration: $i, ${currentOptions.map { case (k, v) => s"$k=$v"}.mkString(", ")}"
-                logMessage(s"Running execution ${q.name} $setup")
-
-                currentExecution = q.name
-                currentPlan = q match {
-                  case query: Query =>
-                    try {
-                      query.newDataFrame().queryExecution.executedPlan.toString()
-                    } catch {
-                      case e: Exception =>
-                        s"failed to parse: $e"
-                    }
-                  case _ => ""
-                }
-                startTime = System.currentTimeMillis()
+      timeout: Long = 0L,
+      resultLocation: String = resultsLocation,
```

Comment:
I was not sure, because `Benchmark` does pass it around as an argument, and it gets overriden in the notebooks.